core/pages/Product.js

```diff
@@ -99,6 +100,7 @@ export default {
           this.loading = false
           this.defaultOfflineImage = this.product.image
           this.onStateCheck()
+          this.$store.dispatch('recently-viewed/addItem', this.product)
```

Comment:
this will not work if we click on any of suggested products since `product` component will remain mounted (in other words hook will not be called). I believe we can add this product automatically for example 
- create some router afterEach hook for this or 
- extend current product store (store.extends property) to dispatch this action each time product is changed
- listen for product object mutation (preferred but you need to take care of variants - we need to push the object from mutation only if the whole product has changed, not only it's variant)